File: examples/python/scan.py
from __future__ import annotations

import sys
from pathlib import Path

import httpx2


def main() -> None:
    image_path = Path("tests/fixtures/images/known_match.bin")
    with httpx2.Client(timeout=10) as client:
        response = client.post(
            "http://localhost:8080/v1/scan",
            files={"image": ("known_match.jpg", image_path.read_bytes(), "image/jpeg")},
        )
    response.raise_for_status()
    _ = sys.stdout.write(f"{response.text}\n")


if __name__ == "__main__":
    main()

File: examples/node/scan.mjs
import { readFile } from "node:fs/promises";

const image = await readFile("tests/fixtures/images/known_match.bin");
const form = new FormData();
form.append("image", new Blob([image], { type: "image/jpeg" }), "known_match.jpg");

const response = await fetch("http://localhost:8080/v1/scan", {
  method: "POST",
  body: form,
});

if (!response.ok) {
  throw new Error(`scan failed: ${response.status}`);
}

console.log(await response.text());
